tests/lib/rules/jsx-no-leaked-render.js

```diff
@@ -232,7 +227,7 @@ ruleTester.run('jsx-no-leaked-render', rule, {
         return (
           <>
             {0 ? <Something/> : null}
-            {'' ? <Something/> : null}
```

Comment:
can we avoid removing this test case entirely? instead, it'd be great if we had one test case that specifies a react version of 17.9999.999 and another that specifies 18.0.0.